synapse/util/hash.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2019 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import hashlib
+
+import unpaddedbase64
+
+
+def sha256_and_url_safe_base64(input_text):
```

Comment:
not entirely sure this is worth the effort over inlining it. If you import the functions rather than the modules, it's pretty short:

```
lookup_value = encode_base64(sha256(to_hash.encode()).digest(), urlsafe=True)
```

Reply:
actually, it's fine. do what you think here.

Reply:
Yeah, I did it because the function name quickly tells you what's going on.